code/go/internal/validator/semantic/validate_capabilities_required.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"fmt"
+	"path"
+	"slices"
+
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/internal/pkgpath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+)
+
+// ValidateCapabilitiesRequired verifies that the required capabilities are added in package manifest
+func ValidateCapabilitiesRequired(fsys fspath.FS) specerrors.ValidationErrors {
+	err := ensureSecurityRulesHasSecurityCapability(fsys)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ensureSecurityRulesHasSecurityCapability(fsys fspath.FS) specerrors.ValidationErrors {
+	securityRuleFilePaths := path.Join("kibana", "security_rule", "*.json")
+	files, err := pkgpath.Files(fsys, securityRuleFilePaths)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredErrorf("error finding Kibana security_rule folder: %w", err)}
+	}
+	if len(files) == 0 {
+		return nil
+	}
+
+	capabilities, err := readCapabilities(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	if !slices.Contains(capabilities, "security") {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredErrorf("found security rule assets in package but security capability is missing in package manifest"),
+		}
+	}
```

Comment:
Should we check that none of the observability capabilities are defined too?https://github.com/elastic/kibana/blob/77102c1854bba7253da5363168e8aafb66a34579/config/serverless.oblt.yml#L145

As this validation is defined now, it just enforces to be present `security` capability.

Reply:
Something like: 
```go
	if slices.Contains(capabilities, "observability") || slices.Contains(capabilities, "apm") || slices.Contains(capabilities, "uptime") {
		return specerrors.ValidationErrors{
			specerrors.NewStructuredErrorf("found security rule assets in package but observability capabilities are defined in package manifest"),
		}
	}
```

The issue here is if those capabilities are modified in Kibana or a new one is added.

Reply:
Or maybe ensure that just one capability is defined, and that one is "security" ? 
Currently, it is just allowed "security":
https://github.com/elastic/kibana/blob/6f4346e4e6f61cf67db50fbe7ce0836c0317cc09/config/serverless.security.yml#L77

That would need to update good_v2 and good_v3 packages (manifests) to just have security capability.

Reply:
Perfect, I agree with that too. Thanks!